qiskit_nature/logging.py

```diff
@@ -0,0 +1,140 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Qiskit Nature Logging Utilities."""
+
+from typing import Optional, Dict, List
+import os
+import sys
+import logging
+
+
+class DefaultStreamHandler(logging.StreamHandler):
+    """Default stream Handler"""
+
+    def __init__(self):
+        """
+        Initialize the handler.
+        """
+        super().__init__(stream=sys.stdout)
+        self.setFormatter(logging.Formatter(fmt="%(asctime)s:%(name)s:%(levelname)s: %(message)s"))
+
+
+class QiskitNatureLogging:
+    """Qiskit Nature Logging."""
+
+    def __init__(self) -> None:
+        self._logging = logging
+
+    def get_levels_for_names(self, names: List[str]) -> Dict[str, int]:
+        """Return logging levels for module names.
+
+        Args:
+            names: list of module names (qiskit_nature.drivers, qiskit.algorithms etc)
+        Returns:
+            Dictionary mapping names to effective level
+        """
+        name_levels: Dict[str, int] = {}
+        for name in names:
+            name_levels[name] = self._logging.getLogger(name).getEffectiveLevel()
+
+        return name_levels
+
+    def set_levels_for_names(
+        self, name_levels: Dict[str, int], add_default_handler: bool = True
+    ) -> Dict[str, int]:
+        """Set logging levels for module names.
+
+        Args:
+            name_levels: Dictionary of module names (qiskit_nature.drivers, qiskit.algorithms etc)
+                         to desired level
+            add_default_handler: add or not the default stream handler
+        Returns:
+            Dictionary mapping names to effective level
+        """
+        names: List[str] = []
+        for name, level in name_levels.items():
+            names.append(name)
+            logger = self._logging.getLogger(name)
+            logger.setLevel(level)
+            logger.propagate = False
+            handlers = logger.handlers
+            if add_default_handler and not any(
+                True for h in handlers if isinstance(h, DefaultStreamHandler)
+            ):
+                self.add_handler(name, DefaultStreamHandler())
+
+        return self.get_levels_for_names(names)
```

Comment:
What's you reason for doing this?

Reply:
I was actually only referring to the `return`-statement. I don't see a good reason for that..

Reply:
I thought useful to return the effective logging level. That indicates exactly the levels for the domains entered in case for some reason the NOTSET is entered, there may still logging set for the domains and the user won't know what level:

getEffectiveLevel

Indicates the effective level for this logger. If a value other than NOTSET has been set using setLevel is returned.
Otherwise, the hierarchy is traversed towards the root until a value other than NOTSET is found, 
and that value is returned. The value returned is an integer, typically one of logging.DEBUG, logging.INFO etc.

Reply:
Hm I see. I was just wondering whether it is a good default for the setter utility to automatically also function as a getter. If there were no public getter I would agree but since there is I would rather follow a Linux-philosophy: no news is good news :smile: 

I don't have a strong opinion on this though I just wanted to discuss your reasoning :slightly_smiling_face:

Reply:
I removed the return value